src/components/uri/Visualization.tsx

```diff
@@ -124,8 +127,18 @@ function fetchVisualizationClass(
     sdk: SDK,
     visualizationClassUri: string
 ): Promise<VisualizationClass.IVisualizationClass> {
-    return sdk.xhr.get(visualizationClassUri)
+    const sdkCache: ISdkCache = getSdkCache();
+    const apiCallIdentifier = `get.${visualizationClassUri}`;
+    const cachedResponse = get<Promise<VisualizationClass.IVisualizationClass>>(
+        sdkCache, apiCallIdentifier
+    );
+    if (cachedResponse) {
+        return cachedResponse;
+    }
+    const promise = sdk.xhr.get(visualizationClassUri)
```

Comment:
I think we should make sure that if the promise would end in `.catch()`, we remove it from the cache so that additional requests would not be blocked when trying to do 'retry'